hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/TokenTupleUtils.java

```diff
@@ -296,14 +300,43 @@ private static Tuple hederaTokenTupleFor(@NonNull final Token token) {
                 expiryTupleFor(token));
     }
 
+    @NonNull
+    private static Tuple hederaTokenTupleForV2(@NonNull final Token token) {
```

Comment:
would it be possible (or wise?) to call `hederaTokenTupleFor` and then just add the metadata key?